x-pack/plugins/gis/public/shared/layers/sources/es_geohashgrid_source/update_source_editor.js

```diff
@@ -45,28 +45,33 @@ export class UpdateSourceEditor extends Component {
     this.setState({ fields: indexPattern.fields });
   }
 
-  onMetricsChange = (metrics) => {
+  _onMetricsChange = (metrics) => {
     this.props.onChange({ propName: 'metrics', value: metrics });
-  }
+  };
 
-  renderMetricsEditor() {
-    if (this.props.renderAs === RENDER_AS.HEATMAP) {
-      return null;
-    }
+  _renderMetricsEditor() {
+
+    const metricsFilter = (this.props.renderAs === RENDER_AS.HEATMAP) ?  ((m) => {
```

Comment:
metricsFilter is optional in MetricsEditor so then is no need for always true filter